tuf/keys.py

```diff
@@ -1016,8 +1053,66 @@ def format_rsakey_from_pem(pem):
   # Ensure the PEM string has a valid header and footer.  Although a simple
   # validation of 'pem' is performed here, a fully valid PEM string is
   # needed to later successfully verify signatures.
-  pem_header = '-----BEGIN PUBLIC KEY-----'
-  pem_footer = '-----END PUBLIC KEY-----'
+  public_pem = extract_pem(pem) 
+
+  # Begin building the RSA key dictionary. 
+  rsakey_dict = {}
+  keytype = 'rsa'
+  public = public_pem 
+
+  # Generate the keyid of the RSA key.  'key_value' corresponds to the
+  # 'keyval' entry of the 'RSAKEY_SCHEMA' dictionary.  The private key
+  # information is not included in the generation of the 'keyid' identifier.
+  key_value = {'public': public,
+               'private': ''}
+  keyid = _get_keyid(keytype, key_value)
+
+  rsakey_dict['keytype'] = keytype
+  rsakey_dict['keyid'] = keyid
+  rsakey_dict['keyval'] = key_value
+
+  return rsakey_dict
+
+
+
+
+
+def extract_pem(pem, private_pem=False):
+  """
+  <Purpose> 
+    Extract only the portion of the pem that include the header and footer, and
+    any leading and trailing characters removed.  The string returned as the
+    following form:
+    
+    '-----BEGIN PUBLIC KEY----- ... -----END PUBLIC KEY-----'
+
+    or
+
+    '-----BEGIN RSA PRIVATE KEY----- ... -----END RSA PRIVATE KEY-----'
+
+  <Arguments>
+    pem:
+      A string in PEM format.
+
+  <Exceptions>
+    tuf.FormatError, if 'pem' is improperly formatted.
+
+  <Side Effects>
+    Only the public and private portion of the PEM is extracted.  Leading or
+    trailing whitespace is not included in the PEM string returned.
+
+  <Returns>
+    A PEM string (excluding leading and trailing newline characters.  
```

Comment:
I.e. it returns `pem_header` + key material + `pem_footer`, right?

(Also, this line is missing a closing `)`.)